app/controllers/api/v1/users/registrations_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Users
+      class RegistrationsController < Devise::RegistrationsController
+        respond_to :json
+
+        before_action :ensure_valid_client
+
+        def create
+          user = User.from_email(user_params[:email], user_params[:password])
+          user.errors.empty? ? render_success(user) : render_error(user)
```

Comment:
The ternary operator is rarely used in Ruby, i.e., it is not idiomatic (and also i would be inconsistent with Guey's implementation):

```suggestion
          if user.errors.empty?
            render_success(user) :
          else
             render_error(user)
          end
```

or using early returns (but less preferred as it can cause weird issues in controllers 😆 ):

```suggestion
          return render_error(user) unless user.errors.empty?
          
          ender_success(user)
```